### PR TITLE
Release version 0.10.0

### PR DESCRIPTION
**Pull request by cnweaver**
## Description

Release for https://github.com/scimma/hop-client/issues/213

## Checklist

* ~[ ] All new functions and classes are documented and adhere to Google doc style (3.8.3-3.8.6 of [this](http://google.github.io/styleguide/pyguide.html#383-functions-and-methods) document)~
* ~[ ] Add/update sphinx documentation with any relevant changes.~
* ~[ ] Add/update pytest-style tests in `/tests`, ensuring sufficient code coverage.~
* [x] `make test` runs without errors.
* [x] `make lint` doesn't give any warnings.
* [x] `make format` doesn't give any code formatting suggestions.
* [x] `make doc` runs without errors and generated docs render correctly.
* [x] Check that CI pipeline run on this PR passes all stages.
* [x] Review signoff by at least one developer.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+### Changed
+- The `hop.io.Producer` can be used to publish to more than one topic with a single instance.
+  To support this capability, the `write` method now accepts an optional `topic` argument to
+  direct to which topic to publish.
+- The error message resulting when attempting to open a stream with ambiguous credentials has been
+  further clarified.
+
+### Added
+- Users can now explicitly stop the `hop.io.Consumer`'s processing loop.
+  (Thanks to @myNameIsPatrick)
```
Comment (manning-ncsa) [inline CHANGELOG.md:19]: Should this be a link with a URL? As it is, the assumption is that it is a GitHub user account?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.10.0] - 2024-07-31
+### Changed
+- The `hop.io.Producer` can be used to publish to more than one topic with a single instance.
+  To support this capability, the `write` method now accepts an optional `topic` argument to
```
Comment (manning-ncsa) [inline CHANGELOG.md:12]: Is this new functionality documented in for example [the quickstart guide](https://github.com/scimma/hop-client/blob/9ddf2ca95b7063254cc7fb3a47ca101fd8ff4802/doc/user/quickstart.rst) or elsewhere?

File: CHANGELOG.md (hunk 1 above)
Comment (cnweaver) [reply CHANGELOG.md:19]: I guess we could make it one; the current form is somewhat arbitrary.

File: CHANGELOG.md (hunk 2 above)
Comment (cnweaver) [reply CHANGELOG.md:12]: There is a section devoted to it: [Publishing to Multiple Topics](https://github.com/scimma/hop-client/blob/9ddf2ca95b7063254cc7fb3a47ca101fd8ff4802/doc/user/stream.rst#publishing-to-multiple-topics)

**Comment by manning-ncsa** [comment]
As I have no more feedback nor concerns, I approve this request.

**Pull request merged**